SingularityUI/app/components/webhooks/Webhooks.jsx

```diff
@@ -10,206 +10,245 @@ import NewWebhookForm from './NewWebhookForm';
-            dataCollection: 'webhooks',
-            data: [{
-                component: PlainText,
-                prop: {
-                    text: webhook.uri
-                }
-            }, {
-                component: PlainText,
-                prop: {
-                    text: Utils.humanizeText(webhook.type)
-                }
-            }, {
-                component: TimeStamp,
-                className: 'hidden-xs',
-                prop: {
-                    timestamp: webhook.timestamp,
-                    display: 'absoluteTimestamp'
-                }
-            }, {
-                component: PlainText,
-                className: 'hidden-xs',
-                prop: {
-                    text: webhook.user || 'N/A'
-                }
-            }, {
-                component: PlainText,
-                prop: {
-                    text: <b>{webhook.queueSize}</b>
-                }
-            }, {
-                component: Link,
-                className: 'hidden-xs actions-column',
-                prop: {
-                    text: <Glyphicon iconClass='trash' />,
-                    onClickFn: () => this.promptDeleteWebhook(webhook),
-                    title: 'Delete',
-                    altText: "Delete this webhook",
-                    overlayTrigger: true,
-                    overlayTriggerPlacement: 'top',
-                    overlayToolTipContent: 'Delete This Webhook',
-                    overlayId: `deleteWebhook${ webhook.id }`
-                }
-            }]
-        }));
-        return data;
-    },
-
-    render() {
-        return <div><div className='row'><div className='col-md-10'><span className='h1'>Webhooks</span></div><div className='col-md-2 button-container'><button className='btn btn-success' alt="Create a new webhook" title="newWebhook" onClick={this.promptNewWebhook}> New Webhook </button></div></div><OldTable defaultRowsPerPage={this.defaultRowsPerPage} rowsPerPageChoices={this.rowsPerPageChoices} tableClassOpts="table-striped" columnHeads={this.webhookColumns()} tableRows={this.getWebhookTableData()} emptyTableMessage='No Webhooks' dataCollection='webhooks' /></div>;
+        deleteWebhook(webhook);
+      }
+    });
+  },
+
+  newWebhook(uri, type) {
+    const data = {
+      uri,
+      type
+    };
+    if (app.user && app.user.attributes.authenticated) {
+      data.user = app.user.attributes.user.id;
     }
+    return $.ajax({
+      url: `${ config.apiRoot }/webhooks`,
+      type: 'POST',
+      contentType: 'application/json',
+      data: JSON.stringify(data),
+      success: () => this.props.collections.webhooks.fetch().done(() => this.forceUpdate())
+    });
+  },
+
+  promptNewWebhook() {
+    const newWebhook = (uri, type) => this.newWebhook(uri, type);
+    return vex.dialog.open({
+      message: "<div class='new-webhook' />",
+      afterOpen: () => {
+        this.validateInput = input => {
+          try {
+            new URL(input);
```

Comment:
This is a linting error because URL isn't assigned to anything, so it's used for its side effect (of throwing an error if the URL provided is bad).

I didn't want to change the functionality, but there's no other way to fix the linting error here.